dts/arm/st/stm32f0.dtsi

```diff
@@ -49,6 +49,60 @@
 			#address-cells = <1>;
 			#size-cells = <1>;
 			reg = <0x48000000 0x1800>;
+
+			gpioa: gpio@48000000 {
```

Comment:
Is there a special reason to put gpio nodes within the pinctrl node besides that it is done in linux?

This just makes the defines longer or needs some extra logic to strip off the extra "PINCTRL" label.
A gpio device may just use the pinctrl/pinmux device but can be seen independent from the pinmux/pinctrl device.

Reply:
One reason is indeed to stay close to Linux.
Other reason is that on stm32 MCUS, gpio and pin control is done at in same IP. GPIO and PIN only differ from alternate function definition.
Then, we can argue that this should be done the other way round and pinctrl might better be inside gpio node. 
Anyway, this is something that we might have to rework inside stm32 zephyr drivers for a better match.  But this was not the purpose here.